tools/shell/embedded_shell.cpp

```diff
@@ -652,6 +655,15 @@ void EmbeddedShell::run() {
         linenoiseHistorySave(path_to_history);
         free(line);
     }
+    if (reQuery) {
+        auto queryResults = processInput(";");
+        if (!queryResults.empty())
+            for (auto& queryResult : queryResults)
+                if (queryResult->isSuccess()) {
+                    printInterrupted = false;
+                    printExecutionResult(*queryResult);
```

Comment:
As we discussed, try to see if we can avoid this repetition. For instance, this is missing the `else` case?
- Detect EOF before line 633 and append `;` there itself
- Refactor out this fragment in a function.